user/signer.go

```diff
@@ -48,3 +48,40 @@ func NewSignerRFC6979WithID(pk ecdsa.PrivateKey, id ID) *SignerRFC6979 {
 func (s SignerRFC6979) UserID() ID {
 	return s.userID
 }
+
+// StaticSigner emulates real sign and contains already precalculated hash.
+// Provides [Signer] interface.
+type StaticSigner struct {
+	neofscrypto.StaticSigner
+	id ID
+}
+
+// NewStaticSigner creates new StaticSigner.
+func NewStaticSigner(scheme neofscrypto.Scheme, sig []byte, pubKey neofscrypto.PublicKey) *StaticSigner {
+	var id ID
+
+	switch t := pubKey.(type) {
+	case *neofsecdsa.PublicKeyRFC6979:
+		id.SetScriptHash((*keys.PublicKey)(t).GetScriptHash())
+	case *neofsecdsa.PublicKey:
+		id.SetScriptHash((*keys.PublicKey)(t).GetScriptHash())
```

Comment:
Same code for both types, no need to duplicate it.

Reply:
Agree, but in one line doesn't work.
```go
	switch t := pubKey.(type) {
	case *neofsecdsa.PublicKeyRFC6979, *neofsecdsa.PublicKey:
		id.SetScriptHash((*keys.PublicKey)(t).GetScriptHash())
	}
```
with _cannot convert t (variable of type neofscrypto.PublicKey) to type *keys.PublicKey_

Reply:
I think this is a `type swith` limitations, like _fallthrough_ for them

Reply:
Removed this func and the code